src/plugins/layout.ts

```diff
@@ -88,9 +88,9 @@ export default class Layout implements IPlugin {
   }
 
   // Add node to the ShadowDom to store initial adjacent node info in a layout and obtain an index
-  private discoverNode(node: Node): ILayoutState {
-    this.shadowDom.insertShadowNode(node, getNodeIndex(node.parentNode), getNodeIndex(node.nextSibling));
-    return createLayoutState(node);
+  private discoverNode(node: Node): INodeInfo {
+    let shadowNode = this.shadowDom.insertShadowNode(node, getNodeIndex(node.parentNode), getNodeIndex(node.nextSibling));
+    return shadowNode.updateInfo();
```

Comment:
>.updateInfo() [](start = 21, length = 13)

Can't we used the cached info? I'm sure I am missing something here.. but also not intuitive why you would want to compute the info on the fly.

Reply:
There is no cached info at this time. Info is not auto-generated on "insertShadowNode" call.

---
In reply to: [182825776](https://github.com/Microsoft/clarity-js/pull/80#discussion_r182825776) [](ancestors = 182825776)